src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -70,6 +74,10 @@ private boolean validPath(Node node, String key) {
         return validString(node, key) && Files.exists(Paths.get(key));
     }
 
+    public synchronized GreengrassV2DataClient getGreengrassV2DataClient() {
+        return greengrassV2DataClient;
+    }
```

Comment:
any chance that a reference to the old client is being used after it is closed?

Reply:
nevermind - I see in the description that was the reason this behavior wasn't added before - so there was a worry about it.

Reply:
Yes it is possible. However I checked all the usages to verify that they will always call get again. At worst, they would fail and then retry after getting a new client. That said, the client can only really change during a deployment which would be after the client has been used.

Reply:
yeah - I just checked that myself after I posted that comment - so it shouldn't have any impact